### DISCUSSION TITLE
Add "rel=me" Mastodon link verification

### DISCUSSION THREAD

**Discussion by MagicLike** [squidfunk/mkdocs-material#4752] [Q&A] [answered]
Maybe this is just a thing which is already possible with vanilla mkdocs, but I did not see a direct option to add it. (Maybe it could be hard-coded?)
Is there anyone who has an idea how to do this?

**Comment by squidfunk** [comment]
I'm sorry, but could you please provide the entire context of your question? I have a Mastodon account but I'm barely using it.

**Reply by MagicLike** [reply]
Mastodon has a possibility to "verify" your website. It scans the landing page of the provided URL and searches for a backlink to the Mastodon account, where the website is linked. When it also finds a "rel=me" as a backlink attribute, it gives the link a green checkmark and verifys that you own it.

https://docs.joinmastodon.org/user/profile/#verification

**Reply by squidfunk** [reply]
Yes, you can put the link anywhere you like. Either in Markdown, or by using [theme extension](https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme), e.g. with a [block override](https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks):

``` jinja
{% extends "base.html" %}

{% block announce %}
  <a href="https://social.example.com/@username" rel="me">Follow me on Mastodon!</a>
{% endblock %}
```

**Reply by MagicLike** [reply]
To be more specific, I want it to be added to the social icons in the footer, to make it available for all users linking their Mastodon account with the icons...

**Reply by squidfunk** [reply]
You can't add the `rel=me` through the default integration, but with theme extension. Override the [`social`](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/social.html) partial.

**Reply by MagicLike** [reply]
I am trying, but it does not seem to work and I can't spot the issue, beacause I just copied the file name from the documentation...

![grafik](https://user-images.githubusercontent.com/82117109/207165572-352b4eb3-9ea8-41f2-8b6d-7c230aad9a03.png)
```
{% extends "social.html" %}

{% block htmla %}
    {{ super() }}
    rel="me"
{% endblock %}
```

**Reply by squidfunk** [reply]
Please read the [customization guide](https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme) and follow the outlined steps.

**Reply by MagicLike** [reply]
oops, spotted my issue now (I think) and I will try to get it to work...

**Reply by MagicLike** [reply]
It might be a bit janky, but it works....
Is there any way to integrate it natively at some point?
![grafik](https://user-images.githubusercontent.com/82117109/207178858-3df7f3ac-e11b-4f0f-a19c-da1ed00d8231.png)

**Reply by MagicLike** [reply]
I just copied the whole thing and added the `me` part...

**Reply by squidfunk** [reply]
> Is there any way to integrate it natively at some point?

Probably not. It's too specific of a case. Additionally, I'd say that a `<link rel="me" href="..." />` would be much more sensical to use. I don't know if Mastodon supports that, but the requirement for having a clickable hyperlink is nonsense.

**Reply by MagicLike** [reply]
> Additionally, I'd say that a `<link rel="me" href="..." />` would be much more sensical to use. I don't know if Mastodon supports that, but the requirement for having a clickable hyperlink is nonsense.

It is actually supported...

![grafik](https://user-images.githubusercontent.com/82117109/207359548-34a3c5cb-7a89-44cc-8a05-5d4daf821226.png)

**Reply by squidfunk** [reply]
Then, integration is even simpler by overriding `main.html`:

```jinja
{% extends "base.html" %}

{% block extrahead %}
  <link href="https://social.example.com/@username" rel="me" />
{% endblock %}
```

This is the recommended way, because using `extrahead` will __never__ interfere with minor or major upgrades.

**Reply by MagicLike** [reply]
Ohhh! Ok, I will try it out and I hope it works... 👍

**Reply by MagicLike** [reply]
That totally worked! Thank you for your help!

**Comment by squidfunk** [comment, marked as answer]
With the latest developments and Melon Tusk going rampage, I believe we should natively support Mastodon verification, which is why the `rel=me` attribute will now be automatically added to social links that use a Mastodon icon. This was added in 3fc8b6be8.

Just add a [social link](https://squidfunk.github.io/mkdocs-material/setup/setting-up-the-footer/#social-links) and add use one of the following icons:

- `:simple-mastodon:`
- `:material-mastodon:`
- `:fontawesome-brands-mastodon:`

**Reply by squidfunk** [reply]
The changes were released as part of [9.0.11](https://github.com/squidfunk/mkdocs-material/releases/tag/9.0.11).

**Reply by Andre601** [reply]
Something I wonder: Why not just apply `rel="me"` in general? Why do this extra check for the Mastodon icon? From what I know is the me rel not really affecting any other site whatsoever, so having it as default would perhaps be better (not to mention gets rid of a somewhat pointless if-check alongside assigning the rel).

**Reply by Andre601** [reply]
Actually considering, it could perhaps be a neat idea for the future to include another setting for the socials to allow people to override the default rel set, so that they could add like "noreferrer" without having to extend the theme and override the socials partial.

I could PR something if that would be an idea. I mean something like this:
```yaml
extra:
  social:
    # Overrides the default rel="noopener" with rel="noopener me"
    - icon: fontawesome/brands/mastodon
      link: https://fosstodon.org/@squidfunk
      rel: 'noopener me'
    # Uses default rel="noopener"
    - icon: fontawesome/brands/github
      link: https://github.com/squidfunk
```
Tho I still think including me by default is preferrable

**Reply by MagicLike** [reply]
I think that is a great idea! Why not including `me`, `noopener` and `noreferrer` by default?

**Reply by Andre601** [reply]
I think that would be a good idea.

**Reply by Guts** [reply]
> With the latest developments and Melon Tusk going rampage, I believe we should natively support Mastodon verification, which is why the `rel=me` attribute will now be automatically added to social links that use a Mastodon icon. This was added in [3fc8b6b](https://github.com/squidfunk/mkdocs-material/commit/3fc8b6be8b926f5ce4dd43c1eff7e0abe7a42e7a).
> 
> Just add a [social link](https://squidfunk.github.io/mkdocs-material/setup/setting-up-the-footer/#social-links) and add use one of the following icons:
> 
>     * `:simple-mastodon:`
> 
>     * `:material-mastodon:`
> 
>     * `:fontawesome-brands-mastodon:`

Is this behaviour compatible with the privacy plugin? https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/?h=noopen#external-links

**Reply by squidfunk** [reply]
All social links have `noopener` set anyway, so I see no problem. Otherwise please elaborate what you think might be problematic.
